### PR TITLE
Implement k8s version picker and fix LB teardown

### PR DESCRIPTION
**Pull request by albinsun**
## Changes
1. **IMPLEMENT** RKE1/RKE2 version picker via config option `k8s-version` (`vX.Y`)
   * No need to specify full `RKE1-version` and `RKE2-version`
   * `k8s-version` act as prefix for filtering available versions.
     > e.g.
     > `v1.26` will get `v1.26.15-rancher1-1` from supports `['v1.25.16-rancher2-3', 'v1.26.15-rancher1-1', 'v1.27.12-rancher1-1']`
   * Fail fast if no matched versions. Help us to fix ASAP if target `k8s-version` is not compatible with current Rancher.
     > e.g.
     > Following previous example, if `k8s-version` is `v1.28`, TC will fail directly since Rancher endpoint do not support at all instead of use hightest supported but non-target `v1.27.12`.

2. **FIX** `409` error hit in daily test by enhancing load balancer teardown
   <details>

   image
   </summary>

3. **UPDATE** openSUSE Leap to 15.5
   * Ref. [[TEST] Upgrade the openSUSE image to 15.5, RKE1 & RKE2 versions to 1.27 in test CI](https://github.com/harvester/tests/issues/1087#issuecomment-2060465519)

## Notes
1. Bundle with **harvester-baremetal-ansible/PR#32** - [Fix ip pool subnet for vlan100](https://github.com/harvester/harvester-baremetal-ansible/pull/32#top) to take effect on daily test

## Verification
* `harvester_e2e_tests/integrations/test_9_rancher_integration.py`
### Daily Env.
* `harvester-v1.3.0`
* `rancher-v2.8.2`: supports K8s `v1.25` ~ `v1.27`
  * RKE2: `['v1.25.13+rke2r1', 'v1.25.15+rke2r2', 'v1.25.16+rke2r1', 'v1.26.8+rke2r1', 'v1.26.10+rke2r2', 'v1.26.11+rke2r1', 'v1.26.13+rke2r1', 'v1.26.14+rke2r1', 'v1.26.15+rke2r1', 'v1.27.5+rke2r1', 'v1.27.7+rke2r2', 'v1.27.8+rke2r1', 'v1.27.10+rke2r1', 'v1.27.11+rke2r1', 'v1.27.12+rke2r1']`
  * RKE1: `['v1.25.16-rancher2-3', 'v1.26.15-rancher1-1', 'v1.27.12-rancher1-1']`

#### Positive TCs
* `k8s-version`: `v1.27`
  * <details>
    <summary> :green_circle: RKE2: Should pick up <code>v1.27.12+rke2r1</code>
    </summary>

    image
    </details>

  * <details>
    <summary> :green_circle: RKE1: Should pick up <code>v1.27.12-rancher1-1</code>
    </summary>

    image
    image
    </details>

  * <details>
    <summary> :green_circle: Test pass
    </summary>
    
    image
    </details>

#### Negtive TCs
1. `k8s-version`: `v1.28` (Too new)
   * <details open>
     <summary> :green_circle: RKE2: Fail due to no matched version
     </summary>
 
     image
     </details>
   
   * <details open>
     <summary> :green_circle: RKE1: Fail due to no matched version
     </summary>
     
     image
     </details>

1. `k8s-version`: `v1.24` (Too old)
   * <details open>
     <summary> :green_circle: RKE2: Fail due to no matched version
     </summary>
 
     image
     </details>
   
   * <details open>
     <summary> :green_circle: RKE1: Fail due to no matched version
     </summary>
     
     image
     </details>

#### Environment
* `harvester-v1.2.2-dev-20240329`
* `rancher-v2.7.10`: supports K8s `v1.23` ~ `v1.26`
   * RKE2: `['v1.23.10+rke2r1', 'v1.23.13+rke2r1', 'v1.23.14+rke2r1', 'v1.23.15+rke2r1', 'v1.23.16+rke2r1', 'v1.23.17+rke2r1', 'v1.24.4+rke2r1', 'v1.24.7+rke2r1', 'v1.24.8+rke2r1', 'v1.24.9+rke2r2', 'v1.24.10+rke2r1', 'v1.24.11+rke2r1', 'v1.24.13+rke2r1', 'v1.24.14+rke2r1', 'v1.24.15+rke2r1', 'v1.24.16+rke2r1', 'v1.24.17+rke2r1', 'v1.25.7+rke2r1', 'v1.25.9+rke2r1', 'v1.25.10+rke2r1', 'v1.25.11+rke2r1', 'v1.25.12+rke2r1', 'v1.25.13+rke2r1', 'v1.25.15+rke2r2', 'v1.25.16+rke2r1', 'v1.26.5+rke2r1', 'v1.26.6+rke2r1', 'v1.26.7+rke2r1', 'v1.26.8+rke2r1', 'v1.26.10+rke2r2', 'v1.26.11+rke2r1', 'v1.26.13+rke2r1', 'v1.26.14+rke2r1', 'v1.26.15+rke2r1']`
   * RKE1: `['v1.23.16-rancher2-3', 'v1.24.17-rancher1-1', 'v1.25.16-rancher2-3', 'v1.26.15-rancher1-1']`

#### Positive TCs
* `k8s-version`: `v1.26`
  * <details>
    <summary> :green_circle: RKE2: Should pick up <code>v1.26.15+rke2r1</code>
    </summary>

    pick-rke2
    </details>

  * <details>
    <summary> :green_circle: RKE1: Should pick up <code>v1.26.15-rancher1-1</code>
    </summary>

    pick-rke1
    </details>

  * <details>
    <summary> :green_circle: Test pass
    </summary>
    
    image
    </details>

* `k8s-version`: `v1.25`
  * :green_circle: RKE2: Should pick up <code>v1.25.16+rke2r1</code>
  * <details>
    <summary> :green_circle: RKE1: Should pick up <code>v1.25.16-rancher2-3</code>
    </summary>

    image
    </details>

  * <details>
    <summary> :green_circle: Test pass
    </summary>
    
    image
    </details>

#### Negtive TCs
1. `k8s-version`: `v1.27` (Too new)
   * <details open>
     <summary> :green_circle: RKE2: Fail due to no matched version
     </summary>
 
     image
     </details>
   
   * <details open>
     <summary> :green_circle: RKE1: Fail due to no matched version
     </summary>
     
     image
     </details>

1. `k8s-version`: `v1.22` (Too old)
   * <details open>
     <summary> :green_circle: RKE2: Fail due to no matched version
     </summary>
 
     image
     </details>
   
   * <details open>
     <summary> :green_circle: RKE1: Fail due to no matched version
     </summary>
     
     image
     </details>

### CODE REVIEW

File: harvester_e2e_tests/fixtures/rancher_api_client.py [hunk 1]
```diff
@@ -20,28 +20,59 @@ def rancher_api_client(request):
...
+
+
+@pytest.fixture(scope='session')
+def rke1_version(request, rancher_api_client):
+    target_ver = request.config.getoption("--k8s-version")
+
+    code, data = rancher_api_client.settings.get("k8s-versions-current")
+    assert 200 == code, (code, data)
+    supported_vers = data["value"].split(",")
+    assert supported_vers
```
Comment (lanfon72) [inline harvester_e2e_tests/fixtures/rancher_api_client.py:51]: There might have a potential bug the value is a empty string like:
```python3
>>> ''.split(",")
['']
```

then the assertion will never failed.
but consider that the following `_pickup_k8s_version` will also raise assertion error, so we can simply remove this one.

**Review by lanfon72 — APPROVED** [review]
In general LGTM

File: apiclient/harvester_api/managers/networks.py [hunk 2]
```diff
@@ -107,3 +107,16 @@ def get(self, name="", *, raw=False):
     def delete(self, name, *, raw=False):
         path = self.PATH_fmt.format(name=f"/{name}", API_VERSION=self.API_VERSION)
         return self._delete(path, raw=raw)
+
+
+class LoadBalancerManager(BaseManager):
+    PATH_fmt = "{API_VERSION}/harvester/loadbalancer.harvesterhci.io.loadbalancers{lb_id}"
+    API_VERSION = "v1"
+
+    def get(self, lb_id="", *, raw=False):
```
[the remark this answers is not in the document]
Comment (albinsun) [reply apiclient/harvester_api/managers/networks.py]: Fixed in commit https://github.com/harvester/tests/pull/1222/commits/b514235a00567f985d13c0fc1624cc11ee9997a4.

File: harvester_e2e_tests/fixtures/rancher_api_client.py [hunk 3]
```diff
@@ -20,28 +20,59 @@ def rancher_api_client(request):
     return api
 
 
+def _pickup_k8s_version(versions: [str], target_version: str):
```
[the remark this answers is not in the document]
Comment (albinsun) [reply harvester_e2e_tests/fixtures/rancher_api_client.py]: Fixed in commit https://github.com/harvester/tests/pull/1222/commits/b514235a00567f985d13c0fc1624cc11ee9997a4.

File: harvester_e2e_tests/fixtures/rancher_api_client.py [hunk 4]
```diff
@@ -20,28 +20,59 @@ def rancher_api_client(request):
...
+    sorted_vers = sorted(versions,
+                         key=lambda v: parse_version(v.split("+")[0].split("-")[0]),
+                         reverse=True)
+
+    for ver in sorted_vers:
+        if ver.startswith(target_version):
+            warnings.warn(UserWarning(f"Adopt {ver} for target k8s-version {target_version}"))
+            return ver
+
+    assert False, (
```
[the remark this answers is not in the document]
Comment (albinsun) [reply harvester_e2e_tests/fixtures/rancher_api_client.py]: Fixed in commit https://github.com/harvester/tests/pull/1222/commits/b514235a00567f985d13c0fc1624cc11ee9997a4.

File: harvester_e2e_tests/fixtures/images.py [hunk 5]
```diff
@@ -20,7 +14,7 @@
 def image_opensuse(request, api_client):
     image_server = request.config.getoption("--image-cache-url")
     url = urlparse(
-        request.config.getoption("--opensuse-image-url") or DEFAULT_OPENSUSE_IMAGE_URL
```
[the remark this answers is not in the document]
Comment (albinsun) [reply harvester_e2e_tests/fixtures/images.py:23]: Config PR for ECM env. is harvester/harvester-baremetal-ansible/pull/32
Will send config PR for Provo on gitlab seperately.

[the remark this answers is not in the document]
Comment (lanfon72) [reply harvester_e2e_tests/fixtures/images.py:23]: Provo (E&I) still using the ansible in gitlab, you might need to create a PR for it.

**Review by lanfon72 — APPROVED** [review]

**Review by TachunLin — APPROVED** [review]
LGTM. 
Tested on Rancher v2.8.2 with Harvester `v1.2-1aa85320-head`

* Passed all single node Harvester Rancher integration test on local environment
  image
  
  image

**Comment by khushboo-rancher** [comment]
@albinsun Can you please help backport this to 1.2 & 1.3

**Pull request merged**